### PR TITLE
native improvements 5

### PR DESCRIPTION
**Pull request by peter-quix**
* Documentation reworked
* Renamed stream.parameters -> stream.timeseries
* Renamed stream.timeseries.on_received to on_data_received
* renamed buffer.on...received to on..released
* renamed CreateTopicConsumer -> GetTopicConsumer
* renamed CreateTopicProducer -> GetTopicProducer 
* renamed some of the leftover .Write to .Publish
* renamed OnMessageRead -> OnMessageReceived
* renamed to/from_pandas_dataframe to to/from_dataframe
* Added InMemoryStorage with optional backing storage such as LocalFileStorage
* exposing topic on the stream in python to simplify callback. This resulted in removal of topic in many python callbacks

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -185,25 +185,24 @@ Once we have setup our producer, it's time to see how to consume data from a top
 import pandas as pd
 
 from quixstreams import *
-from quixstreams.app import App
 
 # Client connecting to Kafka instance locally without authentication. 
 client = KafkaStreamingClient('127.0.0.1:9092')
 
-# Open the input topic where to consume data from.
+# Open the input topic we consume data from.
```
Comment (JotaBlanco) [inline README.md]: Minor thing: Get instead of open to be consistent with sdk

File: docs/app-management.md [hunk 2]
```diff
@@ -16,29 +16,31 @@ Each of these is described in the following sections.
...
 
 ```py
-def read_stream(new_stream: StreamConsumer):
-
-    buffer = new_stream.parameters.create_buffer()
+from quixstreams import TopicConsumer, StreamConsumer
 
-    def on_dataframe_handler(stream: StreamConsumer, df: pd.DataFrame):
-        print(df.to_string())
+def on_stream_received_handler(new_stream: StreamConsumer):
```
Comment (JotaBlanco) [inline docs/app-management.md]: we use new_stream always. In my opinion, that "new" can be a bit misleading. Would we prefer someting linke stream_i?

File: docs/features/data-frames.md [hunk 3]
```diff
@@ -1,29 +1,25 @@
...
-Quix Streams uses Pandas DataFrames just as a representation of the common [TimeseriesData](#timeseriesdata-format) format used to read and write data to Quix.
-
-For example, the following [TimeseriesData](../../read/#timeseriesdata-format):
+For example, the following [TimeseriesData](../consume.md#timeseriesdata-format):
 
 | Timestamp | CarId (tag) | Speed | Gear |
 | --------- | ----------- | ----- | ---- |
 | 1         | car-1       | 120   | 3    |
-| 2         | car-2       | 123   | 3    |
+| 1         | car-2       | 123   | 3    |
```
Comment (JotaBlanco) [inline docs/features/data-frames.md:10]: is this change on purpose? timestamp for rows 1 and 2 are the same now

File: src/CsharpClient/Quix.Streams.ManyStreamTest/StreamingTest.cs [hunk 4]
```diff
@@ -27,10 +27,10 @@ public void Run(CancellationToken ct)
                     streamCounter++;
                     Console.WriteLine($"Stream count: {streamCounter}");
                 };
-                /*var buffer = reader.Parameters.CreateBuffer();
+                /*var buffer = reader.Timeseries.CreateBuffer();
```
Comment (JotaBlanco) [inline src/CsharpClient/Quix.Streams.ManyStreamTest/StreamingTest.cs:30]: Upper case?

File: src/CsharpClient/Quix.Streams.ManyStreamTest/StreamingTest.cs [hunk 5]
```diff
@@ -43,10 +43,10 @@ public void Run(CancellationToken ct)
                 var stream = topicProducer.CreateStream();
                 var data = new Quix.Streams.Streaming.Models.TimeseriesData();
                 data.AddTimestampNanoseconds(10).AddValue("test", DateTime.UtcNow.ToBinary());
-                stream.Parameters.Buffer.Write(data);
+                stream.Timeseries.Buffer.Publish(data);
                 stream.Events.AddTimestampNanoseconds(10).AddValue("test1", "val1");
                 stream.Properties.Location = "/test";
-                stream.Parameters.AddDefinition("test");
+                stream.Timeseries.AddDefinition("test");
```
Comment (JotaBlanco) [inline src/CsharpClient/Quix.Streams.ManyStreamTest/StreamingTest.cs:49]: Same here, upper case?

File: src/CsharpClient/Quix.Streams.PerformanceTest/PerformanceTest.cs [hunk 6]
```diff
@@ -18,14 +18,14 @@ public void Run(CancellationToken ct)
...
 
-            //var topicConsumer = client.CreateTopicConsumer();
-            var topicProducer = client.CreateTopicProducer();
+            //var topicConsumer = client.GetTopicConsumer();
+            var topicProducer = client.GetTopicProducer();
 
             //topicConsumer.OnStreamReceived += (s, stream) =>
             //{
-            //    var buffer = stream.Parameters.CreateBuffer();
+            //    var buffer = stream.Timeseries.CreateBuffer();
```
Comment (JotaBlanco) [inline src/CsharpClient/Quix.Streams.PerformanceTest/PerformanceTest.cs:26]: image
VS 
image

Comment (JotaBlanco) [reply src/CsharpClient/Quix.Streams.PerformanceTest/PerformanceTest.cs:26]: shouldn't this be lowercase? or is that for python only? I won't repeat this comment but there are more instances

File: src/CsharpClient/Quix.Streams.Process.Common.Test/TestTelemetryKafkaConsumer.cs [hunk 7]
```diff
@@ -9,7 +9,7 @@
 namespace Quix.Streams.Process.Common.Test
 {
     /// <summary>
-    /// Kafka Reader that uses a mocked Message broker for test purposes
+    /// Kafka Consumer that uses a mocked Message broker for test purposes
```
Comment (JotaBlanco) [inline src/CsharpClient/Quix.Streams.Process.Common.Test/TestTelemetryKafkaConsumer.cs]: Little detail, but upper or lowercase? Not consistent with next one

File: src/CsharpClient/Quix.Streams.Process.Common.Test/TestTelemetryKafkaProducer.cs [hunk 8]
```diff
@@ -3,7 +3,7 @@
 namespace Quix.Streams.Process.Common.Test
 {
     /// <summary>
-    /// Kafka Writer that uses a mocked Message broker for test purposes
+    /// Kafka producer that uses a mocked Message broker for test purposes
```
Comment (JotaBlanco) [inline src/CsharpClient/Quix.Streams.Process.Common.Test/TestTelemetryKafkaProducer.cs:6]: Previous one and this one not consistent: lower or upper case?

File: src/CsharpClient/Quix.Streams.Process.Samples/RandomDataProducer.cs [hunk 9]
```diff
@@ -9,10 +9,10 @@ namespace Quix.Streams.Process.Samples
     /// <summary>
     /// Simply reader component that generates random tdata messages and writes them to the output.
     /// </summary>
-    public class RandomDataReader : StreamComponent
+    public class RandomDataProducer : StreamComponent
```
Comment (JotaBlanco) [inline src/CsharpClient/Quix.Streams.Process.Samples/RandomDataProducer.cs:12]: Isn't the previous Reader the current Consumer?

File: src/CsharpClient/Quix.Streams.Process.Samples/RandomDataProducer.cs [hunk 10]
```diff
@@ -9,10 +9,10 @@ namespace Quix.Streams.Process.Samples
...
     {
 
-        public RandomDataReader()
+        public RandomDataProducer()
```
Comment (JotaBlanco) [inline src/CsharpClient/Quix.Streams.Process.Samples/RandomDataProducer.cs:15]: Isn't the previous Reader the current Consumer?

File: src/CsharpClient/Quix.Streams.Process/Kafka/TelemetryKafkaProducer.cs [hunk 11]
```diff
@@ -8,7 +8,7 @@
 namespace Quix.Streams.Process.Kafka
 {
     /// <summary>
-    /// Kafka writer component implementation.
+    /// Kafka Producer component implementation.
```
Comment (JotaBlanco) [inline src/CsharpClient/Quix.Streams.Process/Kafka/TelemetryKafkaProducer.cs]: again minor thing: upper(lower case consistency

File: src/PythonClient/src/quixstreams/models/timeseriesdata.py [hunk 12]
```diff
@@ -265,9 +265,9 @@ def _build_row(pdts):
     @staticmethod
     def from_panda_dataframe(data_frame: pd.DataFrame, epoch: int = 0) -> 'TimeseriesData':
         """
-        Converts Panda DataFrame to TimeseriesData
+        Converts pandasDataFrame to TimeseriesData
```
Comment (JotaBlanco) [inline src/PythonClient/src/quixstreams/models/timeseriesdata.py]: minor thing: pandas DataFrame looks better

File: src/PythonClient/src/quixstreams/models/timeseriesdata.py [hunk 13]
```diff
@@ -265,9 +265,9 @@ def _build_row(pdts):
...
 
-        :param data_frame: The Panda DataFrame to convert to TimeseriesData
+        :param data_frame: The pandasDataFrame to convert to TimeseriesData
```
Comment (JotaBlanco) [inline src/PythonClient/src/quixstreams/models/timeseriesdata.py]: minor thing: pandas DataFrame looks better

File: src/PythonClient/src/quixstreams/models/timeseriesdata.py [hunk 14]
```diff
@@ -178,19 +178,19 @@ def timestamps(self, timestamp_list: List[TimeseriesDataTimestamp]) -> None:
...
         self.__wrapped.Timestamps = dotnet_list
 
-    def to_panda_dataframe(self) -> pd.DataFrame:
+    def to_dataframe(self) -> pd.DataFrame:
         """
-        Converts TimeseriesData to Panda DataFrame
+        Converts TimeseriesData to pandas DataFrame
 
-        :return: Converted Panda DataFrame
+        :return: Converted pandas DataFrame
```
Comment (JotaBlanco) [inline src/PythonClient/src/quixstreams/models/timeseriesdata.py:193]: this is not consistent with following lines (like in 268) where pandasDataFrame is used. Which one do we go for?

File: src/PythonClient/src/quixstreams/models/timeseriesdataraw.py [hunk 15]
```diff
@@ -131,11 +131,11 @@ def to_panda_dataframe(self) -> pd.DataFrame:
         return pd.DataFrame(rows, columns=headers)
 
     @staticmethod
-    def from_panda_dataframe(data_frame: pd.DataFrame, epoch: int = 0) -> 'TimeseriesDataRaw':
+    def from_dataframe(data_frame: pd.DataFrame, epoch: int = 0) -> 'TimeseriesDataRaw':
         """
-        Converts Panda DataFrame to TimeseriesData
+        Converts from pandas DataFrame to TimeseriesDataRaw
```
Comment (JotaBlanco) [inline src/PythonClient/src/quixstreams/models/timeseriesdataraw.py:136]: same pandasDataFrame VS pandas Dataframe debate

File: src/PythonClient/src/quixstreams/models/timeseriesdataraw.py [hunk 16]
```diff
@@ -80,11 +80,11 @@ def __str__(self):
...
         return text
 
-    def to_panda_dataframe(self) -> pd.DataFrame:
+    def to_dataframe(self) -> pd.DataFrame:
         """
-        Converts TimeseriesDataRaw to Panda DataFrame
+        Converts TimeseriesDataRaw to pandasDataFrame
 
-        :return: Converted Panda DataFrame
+        :return: Converted pandasDataFrame
```
Comment (JotaBlanco) [inline src/PythonClient/src/quixstreams/models/timeseriesdataraw.py]: same pandasDataFrame VS pandas Dataframe debate

File: src/PythonClient/src/quixstreams/models/timeseriesdataraw.py [hunk 17]
```diff
@@ -131,11 +131,11 @@ def to_panda_dataframe(self) -> pd.DataFrame:
...
 
     @staticmethod
-    def from_panda_dataframe(data_frame: pd.DataFrame, epoch: int = 0) -> 'TimeseriesDataRaw':
+    def from_dataframe(data_frame: pd.DataFrame, epoch: int = 0) -> 'TimeseriesDataRaw':
         """
-        Converts Panda DataFrame to TimeseriesData
+        Converts from pandas DataFrame to TimeseriesDataRaw
 
-        :param data_frame: The Panda DataFrame to convert to TimeseriesData
+        :param data_frame: The pandas DataFrame to convert to TimeseriesData
```
Comment (JotaBlanco) [inline src/PythonClient/src/quixstreams/models/timeseriesdataraw.py:138]: same pandasDataFrame VS pandas Dataframe debate

File: src/PythonClient/src/quixstreams/models/timeseriesdataraw.py [hunk 18]
```diff
@@ -154,7 +154,7 @@ def from_panda_dataframe(data_frame: pd.DataFrame, epoch: int = 0) -> 'Timeserie
             possible_time_vals = data_frame.select_dtypes(include=['int', 'int64'])
             if possible_time_vals.count()[0] == 0:
                 raise Exception(
-                    "Panda data frame does not contain a suitable time column. Make sure to label the column 'time' or 'timestamp', else first integer column will be picked up as time")
+                    "pandas data frame does not contain a suitable time column. Make sure to label the column 'time' or 'timestamp', else first integer column will be picked up as time")
```
Comment (JotaBlanco) [inline src/PythonClient/src/quixstreams/models/timeseriesdataraw.py]: same pandasDataFrame VS pandas Dataframe debate

**Review by JotaBlanco — COMMENTED** [review]
@peter-quix check the feedback, then I'll approve

File: README.md (hunk 1 above)
Comment (JotaBlanco) [reply README.md]: I didn't check if this was happening anywhere else

File: docs/app-management.md (hunk 2 above)
Comment (JotaBlanco) [reply docs/app-management.md]: or stream_received

File: docs/features/data-frames.md (hunk 3 above)
Comment (JotaBlanco) [reply docs/features/data-frames.md:10]: same happens in later examples so I assume it's on purpose

Comment (peter-quix) [reply docs/features/data-frames.md:10]: Yes, just wanted to signify the tag being a unique key, resulting in keeping the timestamp twice

File: src/CsharpClient/Quix.Streams.ManyStreamTest/StreamingTest.cs (hunk 4 above)
Comment (peter-quix) [reply src/CsharpClient/Quix.Streams.ManyStreamTest/StreamingTest.cs:30]: C#

File: src/CsharpClient/Quix.Streams.PerformanceTest/PerformanceTest.cs (hunk 6 above)
Comment (peter-quix) [reply src/CsharpClient/Quix.Streams.PerformanceTest/PerformanceTest.cs:26]: c# upper, python lower

File: src/CsharpClient/Quix.Streams.Process.Samples/RandomDataProducer.cs (hunk 10 above)
Comment (peter-quix) [reply src/CsharpClient/Quix.Streams.Process.Samples/RandomDataProducer.cs:15]: This generates data to an output, so I thought producer is more accurate. This could go either way, depending on context tbh

File: src/PythonClient/src/quixstreams/models/timeseriesdataraw.py (hunk 16 above)
Comment (peter-quix) [reply src/PythonClient/src/quixstreams/models/timeseriesdataraw.py]: thought I found all these, but apparently not. Just a typo, fixing

File: src/PythonClient/src/quixstreams/models/timeseriesdataraw.py (hunk 18 above)
Comment (JotaBlanco) [inline src/PythonClient/src/quixstreams/models/timeseriesdataraw.py]: Should be pandas DataFrame

**Review by JotaBlanco — DISMISSED** [review]

**Review by JotaBlanco — APPROVED** [review]
